python/pyarrow/parquet/__init__.py

```diff
@@ -2258,11 +2269,13 @@ class _ParquetDatasetV2:
     1       4  Horse  2022
     """
 
-    def __init__(self, path_or_paths, filesystem=None, filters=None,
+    def __init__(self, path_or_paths, filesystem=None, *, filters=None,
                  partitioning="hive", read_dictionary=None, buffer_size=None,
                  memory_map=False, ignore_prefixes=None, pre_buffer=True,
                  coerce_int96_timestamp_unit=None, schema=None,
-                 decryption_properties=None, **kwargs):
+                 decryption_properties=None, thrift_string_size_limit=None,
+                 thrift_container_size_limit=None,
```

Comment:
I think you need to add it to the signature of `class ParquetDataset.__new__` as well, to pass it through

Reply:
Although if the tests pass ..

Reply:
Yeah, `pq.read_table` invokes dataset directly.

Reply:
Ah, yes, I see